### PR TITLE
Move LoginModal username and password into instance properties, remove refs

### LINKED ISSUE #965: HTTP basic authentication dialog not passing username and password
I confirm (by marking "x" in the [ ] below: [x]):

- [x] This is not a troubleshooting question. [Troubleshooting questions go here: http://www.mattermost.org/troubleshoot/](http://www.mattermost.org/troubleshoot/).
- [x] This doesn't reproduce on web browsers (such as in Chrome). If it does, [issue reports go to the Mattermost Server repository](https://github.com/mattermost/platform/issues).
- [x] I have read [contributing guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md).

---

**Summary**
Attempting to use an HTTP proxy which requires basic authentication fails. From investigating with developer tools, it appears that the username and password are being passed as `undefined` into the LoginModal onLogin function.

**Environment**
- Operating System: macOS
- Mattermost Desktop App version: 4.2.1
- Mattermost Server version: 5.9.0

**Steps to reproduce**
1. Configure an HTTP proxy which requires basic authentication
2. Open the desktop app and try to connect to a server through the proxy
3. Enter username and password when the proxy authentication dialog appears

**Expected behavior**
Authentication succeeds, Mattermost loads

**Observed behavior**
Authentication dialog disappears, "Cannot connect to Mattermost" screen is displayed with ERR_TUNNEL_CONNECTION_FAILED (-111)

**Possible fixes**
Possible regression from 5091f59

### PR DESCRIPTION
**Pull request by MikeNicholls**
Before submitting, please confirm you've
 - [x] read and understood our [Contributing Guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md)
 - [x] completed [Mattermost Contributor Agreement](http://www.mattermost.org/mattermost-contributor-agreement/)
 - [x] executed `npm run lint:js` for proper code formatting

Please provide the following information:

**Summary**
Resolves an issue where username and password were not being passed for HTTP basic authentication.

**Issue link**
Fixes #965

**Test Cases**

**Additional Notes**

### CODE REVIEW

File: src/browser/components/LoginModal.jsx [hunk 1]
```diff
@@ -9,17 +9,16 @@ export default class LoginModal extends React.Component {
   constructor(props) {
     super(props);
     this.handleSubmit = this.handleSubmit.bind(this);
-    this.usernameRef = React.createRef();
-    this.passwordRef = React.createRef();
+    this.state = {
```
Comment (enahum) [inline src/browser/components/LoginModal.jsx]: There is no need to store this values in the component state, they can be class properties
`this.username` and `this.password`

File: src/browser/components/LoginModal.jsx [hunk 2]
```diff
@@ -54,7 +53,10 @@ export default class LoginModal extends React.Component {
                 <FormControl
                   type='text'
                   placeholder='User Name'
-                  ref={this.usernameRef}
+                  value={this.state.username}
```
Comment (enahum) [inline src/browser/components/LoginModal.jsx]: the FormControl does not have to be re-rendered every time there is a key stroke

File: src/browser/components/LoginModal.jsx [hunk 3]
```diff
@@ -54,7 +53,10 @@ export default class LoginModal extends React.Component {
...
+                  onChange={(e) => {
```
Comment (enahum) [inline src/browser/components/LoginModal.jsx]: lets make this a function in the class something like
```js
setUser = (e) => { this.username = e.target.value }
```

then the `onChange` prop is `onChange=this.setUser`

File: src/browser/components/LoginModal.jsx [hunk 4]
```diff
@@ -70,7 +72,10 @@ export default class LoginModal extends React.Component {
                 <FormControl
                   type='password'
                   placeholder='Password'
-                  ref={this.passwordRef}
+                  value={this.state.password}
+                  onChange={(e) => {
```
Comment (enahum) [inline src/browser/components/LoginModal.jsx]: see above comment about the function

File: src/browser/components/LoginModal.jsx (hunk 1 above)
Comment (MikeNicholls) [reply src/browser/components/LoginModal.jsx]: Thanks for the feedback!  I don't have any React experience, so it's much appreciated.  I've replaced the state two with properties.

File: src/browser/components/LoginModal.jsx (hunk 2 above)
Comment (MikeNicholls) [reply src/browser/components/LoginModal.jsx]: Thanks, removed

File: src/browser/components/LoginModal.jsx (hunk 3 above)
Comment (MikeNicholls) [reply src/browser/components/LoginModal.jsx]: Done

File: src/browser/components/LoginModal.jsx (hunk 4 above)
Comment (MikeNicholls) [reply src/browser/components/LoginModal.jsx]: Done

**Comment by amyblais** [comment]
Not sure which dev is owning reviews on this, assigning one more reviewer.

**Pull request merged**